src/redux/reducers/index.js

```diff
@@ -43,15 +44,55 @@ export const resourceTypesReducer = (state = preloadedResourceIdsGroupedByType,
           }
         }
         if (!acc[resourceType]) {
-          acc[resourceType] = new Set();
+          acc[resourceType] = {};
         }
-        if (acc[resourceType].has(resource.id)) {
-          console.warn(`${resourceType} already contains ${id}`); // eslint-disable-line no-console
+        if (!acc[resourceType][subType]) {
+          acc[resourceType][subType] = new Set();
+        }
+        if (acc[resourceType][subType].has(resource.id)) {
+          console.warn(`${resourceType}--${subType} already contains ${id}`); // eslint-disable-line no-console
         } else {
-          acc[resourceType].add(resource.id);
+          acc[resourceType][subType].add(resource.id);
+        }
+        if (!acc[resourceType].total) {
+          acc[resourceType].total = 0;
         }
+        acc[resourceType].total += 1;
+        return acc;
+      }, {});
+    }
+    default:
+      return state;
+  }
+};
+
+const preloadResourceTypeFilters = {};
+export const resourceTypeFiltersReducer = (state = preloadResourceTypeFilters, action) => {
+  switch (action.type) {
```

Comment:
should probably add a case for `actionTypes.CLEAR_PATIENT_DATA`, and return `preloadResourceTypeFilters`